format.swift

```diff
@@ -0,0 +1,548 @@
+      .deletingLastPathComponent()
+  }
+
+  /// The directory in which swift-format should be built.
+  static var swiftFormatBuildDir: URL {
+    packageDir
+      .appendingPathComponent(".swift-format-build")
+  }
+
+  private static var envSearchPaths: [URL] {
+    // Compute search paths from PATH variable.
+    #if os(Windows)
+    let pathSeparator: Character = ";"
+    #else
+    let pathSeparator: Character = ":"
+    #endif
+    return (paths ?? "")
+      .split(separator: pathSeparator)
+      .map(String.init)
+      .compactMap { pathString in
+        return URL(fileURLWithPath: pathString)
+      }
+  }
+
+  private static var paths: String? {
+    #if os(Windows)
+    let pathArg = "Path"
+    #else
+    let pathArg = "PATH"
+    #endif
+
+    return ProcessInfo.processInfo.environment[pathArg]
+  }
+
+  enum ExecutableLookupError: Error, CustomStringConvertible {
+    case notFound(executableName: String)
+
+    var description: String {
+      switch self {
+      case .notFound(executableName: let executableName):
+        return "Executable \(executableName) not found in PATH"
+      }
+    }
+  }
+
+  private static func lookupExecutable(for filename: String) throws -> URL {
+    let executable = envSearchPaths.map { $0.appendingPathComponent(filename) }
+      .first(where: { $0.isExecutableFile })
+    guard let executable else {
+      throw ExecutableLookupError.notFound(executableName: filename)
+    }
+    return executable
+  }
+}
+
+extension URL {
+  var isExecutableFile: Bool {
+    return (self.isFile(path) || self.isSymlink(path)) && FileManager.default.isExecutableFile(atPath: path)
+  }
+
+  private func isFile(_ path: String) -> Bool {
+    let attrs = try? FileManager.default.attributesOfItem(atPath: path)
+    return attrs?[.type] as? FileAttributeType == .typeRegular
+  }
+
+  private func isSymlink(_ path: String) -> Bool {
+    let attrs = try? FileManager.default.attributesOfItem(atPath: path)
+    return attrs?[.type] as? FileAttributeType == .typeSymbolicLink
+  }
+}
+
+// MARK: - Process Runner
+
+/// Keeps track of subprocesses spawned by this script and forwards SIGINT
+/// signals to them.
+class SigIntListener {
+  /// The subprocesses spawned by this script that are currently running.
+  static var runningSubprocesses: Set<Process> = []
+
+  /// Whether a `SIGINT` signal has been received by this script.
+  static var hasReceivedSigInt: Bool = false
+
+  /// Registers a `SIGINT` signal handler that forwards `SIGINT` to all
+  /// subprocesses that are registered in `runningSubprocesses`
+  static func registerSigIntSubprocessTerminationHandler() {
+    #if canImport(Darwin) || canImport(Glibc)
+    signal(SIGINT) { _ in
+      SigIntListener.hasReceivedSigInt = true
+      for process in SigIntListener.runningSubprocesses {
+        kill(process.processIdentifier, SIGINT)
+      }
+    }
+    #endif
+  }
+}
+
+/// Provides convenience APIs for launching and gathering output from a subprocess
```

Comment:
I'm assuming just a copy paste, so haven't actually checked this. FWIW this makes me very sad :(